crates/core/src/entry/validation_dependencies.rs

```diff
@@ -0,0 +1,114 @@
+use crate::network::entry_with_header::EntryWithHeader;
+use holochain_core_types::entry::Entry;
+use holochain_persistence_api::cas::content::Address;
+
+pub trait ValidationDependencies {
+    fn get_validation_dependencies(&self) -> Vec<Address>;
+}
+
+impl ValidationDependencies for EntryWithHeader {
+    fn get_validation_dependencies(&self) -> Vec<Address> {
+        match &self.entry {
+            Entry::App(_, _) => {
+                // an app entry is dependent on its previous header being validated
```

Comment:
Why? I don‘t see why this would be true in the general case..

Reply:
Right, so without doing a call into the zome to retrieve the ValidationPackageDefinition it is impossible to know if it needs just the entry or the full chain. This assumes it is always full chain to support all cases but possibly at the expense of validating entries in a different, less efficient order